src/Engine/Managers/CameraManager/CameraManager.hpp

```diff
@@ -0,0 +1,119 @@
+#ifndef RADIUMENGINE_CAMERAMANAGER_H
+#define RADIUMENGINE_CAMERAMANAGER_H
+
+#include <Engine/Managers/CameraManager/CameraStorage.hpp>
+#include <Engine/RaEngine.hpp>
+#include <Engine/Renderer/RenderTechnique/RenderParameters.hpp>
+#include <Engine/Renderer/RenderTechnique/RenderTechnique.hpp>
+#include <Engine/Renderer/Renderer.hpp>
+#include <Engine/System/System.hpp>
+
+#include <memory>
+
+namespace Ra {
+namespace Engine {
+class RenderObject;
+} // namespace Engine
+} // namespace Ra
+
+namespace Ra {
+namespace Engine {
+
+/**
+ * Interface providing functions to manage a group or type of Cameras
+ * in a specific way.
+ */
+class RA_ENGINE_API CameraManager : public System {
+    // TODO (Mathias) make Camera manager compatible with range for ...
+  public:
+    /// Constructor
+    CameraManager();
+
+    // Make copies impossible
+    CameraManager( const CameraManager& ) = delete;
+    CameraManager& operator=( const CameraManager& ) = delete;
+
+    /// Virtual destructor
+    virtual ~CameraManager();
+
+    /// Get a pointer to the cam-th Camera.
+    virtual const Camera* getCamera( size_t cam ) const = 0;
+
+    /// Add a Camera to the manager ...
+    virtual void addCamera( Camera* cam ) = 0;
+
+    //
+    // Calls for the Renderer
+    //
+
+    /**
+     * @brief Number of Cameras.
+     * This is still a work in progress. The idea is to make it possible for a
+     * CameraManager to tell it has only one Camera, for example if it wants to send
+     * a lot of sources at once in a single RenderParams, let's say a texture.
+     */
+    virtual size_t count() const;
+
+    /**
+     * @brief Call before a render, update the general state of the CameraManager.
+     */
```

Comment:
Actually, it registers the RenderData for rendering through CameraManager::render (only the dt attribute though since the matrices are updated from the requested camera)